src/lib/pure/ketrew_yarn.ml

```diff
@@ -0,0 +1,249 @@
+(**************************************************************************)
+(*  Copyright 2015, Sebastien Mondet <seb@mondet.org>                     *)
+(*                                                                        *)
+(*  Licensed under the Apache License, Version 2.0 (the "License");       *)
+(*  you may not use this file except in compliance with the License.      *)
+(*  You may obtain a copy of the License at                               *)
+(*                                                                        *)
+(*      http://www.apache.org/licenses/LICENSE-2.0                        *)
+(*                                                                        *)
+(*  Unless required by applicable law or agreed to in writing, software   *)
+(*  distributed under the License is distributed on an "AS IS" BASIS,     *)
+(*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or       *)
+(*  implied.  See the License for the specific language governing         *)
+(*  permissions and limitations under the License.                        *)
+(**************************************************************************)
+
+
+open Ketrew_pervasives
+
+open Ketrew_long_running_utilities
+
+open Ketrew_gen_yarn_v0.Distributed_shell_parameters
+open Ketrew_gen_yarn_v0.Run_parameters
+open Ketrew_gen_yarn_v0.Running
+open Ketrew_gen_yarn_v0.Created
+
+include Json.Make_versioned_serialization
+    (Ketrew_gen_yarn_v0.Run_parameters)
+    (Ketrew_gen_versioned.Yarn_run_parameters)
+
+type run_parameters = Ketrew_gen_yarn_v0.Run_parameters.t
+
+let name = "yarn-cluster"
+
+let default_distributed_shell_jar =
+  "/opt/cloudera/parcels/CDH/lib/hadoop-yarn/hadoop-yarn-applications-distributedshell.jar"
+
+let distributed_shell_program
+    ?(hadoop_bin="hadoop")
+    ?(distributed_shell_shell_jar=default_distributed_shell_jar)
+    ~container_memory
+    ~timeout
+    ~application_name
+    program =
+  `Distributed_shell (
+    {hadoop_bin; distributed_shell_shell_jar;
+     container_memory; timeout; application_name}, program)
+
+let create
+    ?(host=Ketrew_host.tmp_on_localhost)
+    ?(daemonize_using=`Python_daemon)
+    ?(daemon_start_timeout=3600.)
+    program =
+  let created = {host; program; daemonize_using; daemon_start_timeout} in
+  `Long_running (name, `Created created |> serialize)
+
+
+let hack_to_string = function
```

Comment:
Yes, I didn't want to expose it in `ketrew_daemonize.mli` to make it less confusing. Should it be shared?